tools/slicec-cs/src/encoding.rs

```diff
@@ -351,9 +355,10 @@ fn encode_action_body(
     let mut code = CodeBlock::default();
     let is_optional = type_ref.is_optional && !is_tagged;
 
-    let value = match (is_optional, type_ref.is_value_type()) {
-        (true, false) => "value!",
-        (true, true) => "value!.Value",
+    let value = match (is_optional, type_ref.is_value_type(), type_ref.is_reference_type()) {
```

Comment:
I didn't change the logic of this code in this PR, even though it's definitely inelegant.

We compute this value here and then use "value" or value in the code below, which is pretty confusing. Not having any comment doesn't help either.